src/transformers/modeling_funnel.py

```diff
@@ -0,0 +1,1529 @@
+# coding=utf-8
+# Copyright 2020-present Google Brain and Carnegie Mellon University Authors and the HuggingFace Inc. team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" PyTorch Funnel Transformer model. """
+
+import os
+from dataclasses import dataclass
+from typing import Optional, Tuple
+
+import numpy as np
+import torch
+from torch import nn
+from torch.nn import CrossEntropyLoss, MSELoss
+from torch.nn import functional as F
+
+from .activations import ACT2FN
+from .configuration_funnel import FunnelConfig
+from .file_utils import (
+    ModelOutput,
+    add_code_sample_docstrings,
+    add_start_docstrings,
+    add_start_docstrings_to_callable,
+    replace_return_docstrings,
+)
+from .modeling_outputs import (
+    BaseModelOutput,
+    MaskedLMOutput,
+    MultipleChoiceModelOutput,
+    QuestionAnsweringModelOutput,
+    SequenceClassifierOutput,
+    TokenClassifierOutput,
+)
+from .modeling_utils import PreTrainedModel
+from .utils import logging
+
+
+logger = logging.get_logger(__name__)
+
+_CONFIG_FOR_DOC = "FunnelConfig"
+_TOKENIZER_FOR_DOC = "FunnelTokenizer"
+
+FUNNEL_PRETRAINED_MODEL_ARCHIVE_LIST = [
+    "funnel_small",  # B4-4-4H768
+    "funnel_small-base",  # B4-4-4H768, no decoder
+    # "funnel-medium",      # B6-3x2-3x2H768
+    # "funnel-medium-base", # B6-3x2-3x2H768, no decoder
+    # "funnel",             # B6-6-6H768
+    # "funnel-base",        # B6-6-6H768, no decoder
+    # "funnel-large",       # B8-8-8H1024
+    # "funnel-large-base",  # B8-8-8H1024, no decoder
+    # "funnel-xlarge-base", # B10-10-10H1024
+    # "funnel-xlarge",      # B10-10-10H1024, no decoder
+]
+
+
+FunnelLayerNorm = nn.LayerNorm
+
+INF = 1e6
```

Comment:
usually we use 10000.0 for the attention mask. Do we have to use 10^6 here? I think this could lead to problems with fp16 because 
```
torch.tensor([10**6]).to(torch.float16)
```
yields inf, while 
```
torch.tensor([10**4]).to(torch.float16)
```
does not

Reply:
This will probably change the results with respect to the original implementation though. Maybe not sufficiently for us to care?

Reply:
Side question, why are infs bad since it goes in a softmax anyway?

Reply:
If it changes the results with the original implementation I think we should keep 10**6 - yeah I guess as long as not all values are `inf` inf that goes into a softmax should be fine (think distillbert) does the same